ssl/t1_lib.c

```diff
@@ -4597,7 +4626,19 @@ int tls_choose_sigalg(SSL_CONNECTION *s, int fatalerrs)
                  * cert type
                  */
                 for (i = 0; i < s->shared_sigalgslen; i++) {
+                    int dtls, minversion, maxversion;
+
+                    /* Check the sigalg version bounds */
                     lu = s->shared_sigalgs[i];
+                    dtls = SSL_CONNECTION_IS_DTLS(s);
+                    minversion = dtls ? lu->mindtls : lu->mintls;
+                    maxversion = dtls ? lu->maxdtls : lu->maxtls;
+                    if (minversion != 0
+                        && ssl_version_cmp(s, minversion, s->version) > 0)
+                        continue;
+                    if (maxversion != 0
+                        && ssl_version_cmp(s, maxversion, s->version) < 0)
+                        continue;
```

Comment:
Do we need to call `tls12_sigalg_allowed`?

Reply:
Same answer as above.